### PR TITLE
windows compilation fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: test/RunChromaprint.cxx [hunk 1]
```diff
@@ -53,7 +54,7 @@ ParseCommandLine(int argc, char **argv)
 	while (auto o = option_parser.Next()) {
 		switch (Option(o.index)) {
 		case OPTION_CONFIG:
-			c.config_path = Path::FromFS(o.value);
+			c.config_path = FromNarrowPath(o.value);
```
Comment (MaxKellermann) [inline test/RunChromaprint.cxx:57]: use-after-free bug

**Comment by MaxKellermann** [comment]
Sigh, @neheb you keep posting low-effort PRs without even testing them. If it compiles, it doesn't mean it's good.

File: src/decoder/plugins/Mpg123DecoderPlugin.cxx [hunk 2]
```diff
@@ -43,14 +44,14 @@ mpd_mpg123_finish() noexcept
...
+mpd_mpg123_open(mpg123_handle *handle, Path path_fs,
 		AudioFormat &audio_format)
 {
-	int error = mpg123_open(handle, path_fs);
+	int error = mpg123_open(handle, NarrowPath(path_fs));
 	if (error != MPG123_OK) {
 		FmtWarning(mpg123_domain,
 			   "libmpg123 failed to open {}: {}",
-			   path_fs, mpg123_plain_strerror(error));
+			   NarrowPath(path_fs).c_str(), mpg123_plain_strerror(error));
```
Comment (MaxKellermann) [inline src/decoder/plugins/Mpg123DecoderPlugin.cxx:54]: Why oh why???

File: test/RunChromaprint.cxx (hunk 1 above)
Comment (neheb) [reply test/RunChromaprint.cxx:57]: Fixed I think?

File: src/decoder/plugins/Mpg123DecoderPlugin.cxx (hunk 2 above)
Comment (neheb) [reply src/decoder/plugins/Mpg123DecoderPlugin.cxx:54]: Fixed.

**Comment by neheb** [comment]
want MSYS2 CI update in this PR?

libupnp status: https://github.com/pupnp/pupnp/pull/436

**Pull request merged**